### PR TITLE
[SQL] Support for PARSE_* functions for time-related types

### LINKED ISSUE #2275: [SQL] format_time for specifier %y returns 4-digit year
The big-query specification of this function requires the non-padded year

### LINKED ISSUE #2716: [SQL] Support PARSE_DATE, PARSE_DATETIME, PARSE_TIME

### PR DESCRIPTION
**Pull request by mihaibudiu**
Fixes #2275 
Fixes #2716

### CODE REVIEW

**Comment by mihaibudiu** [comment]
This PR is mostly Rust and documentation, so it should be easy to review.

File: crates/sqllib/src/timestamp.rs [hunk 1]
```diff
@@ -813,6 +814,36 @@ pub fn hop_TimestampN_ShortInterval_ShortInterval_ShortInterval(
     }
 }
 
+#[doc(hidden)]
+pub fn parse_timestamp__(format: String, st: String) -> Option<Timestamp> {
```
Comment (ryzhyk) [inline crates/sqllib/src/timestamp.rs:818]: Is this one of those cases where you could pass strings by reference, as `&str`?

File: docs/sql/datetime.md [hunk 2]
```diff
@@ -363,59 +363,114 @@ result.  A conjunction of such terms is also accepted if all terms
...
+| `PARSE_TIMESTAMP`  | string_format, string | TIMESTAMP | `PARSE_TIMESTAMP("%Y-%m", '2020-10')` => `2020-10-01 00:00:00` |
+
+If the string cannot be parsed according to the specified format:
+
+- a runtime error is produced if the format is incorrect for producing
+  a value of the required type
+- the result is `NULL` if string cannot be parsed according to the format
+
+For the parsing functions the format string must have enough elements
+to make obtain all elements required for building the result.  For
```
Comment (ryzhyk) [inline docs/sql/datetime.md:384]:
```suggestion
to obtain all elements required for building the result.  For
```

File: docs/sql/datetime.md [hunk 3]
```diff
@@ -363,59 +363,114 @@ result.  A conjunction of such terms is also accepted if all terms
...
+error, since the `%I` element does not have enough information to
+infer whether the hour is AM or PM (an extra `%p` element is needed).
+
+The format string recognizes the following format specifiers; this
+definition follows the Rust `strftime` spec:
+https://docs.rs/chrono/latest/chrono/format/strftime/index.html
+
+## Date Specifiers
+
+|Spec. | Example | Description |
```
Comment (ryzhyk) [inline docs/sql/datetime.md:395]: Do these tables come from Rust docs?

**Review by ryzhyk — APPROVED** [review]

File: docs/sql/datetime.md (hunk 3 above)
Comment (mihaibudiu) [reply docs/sql/datetime.md:395]: yes, copy-paste-edit

File: crates/sqllib/src/timestamp.rs (hunk 1 above)
Comment (mihaibudiu) [reply crates/sqllib/src/timestamp.rs:818]: So far I am never passing strings by reference, but would be glad to discuss a design that would be general.

**Pull request merged**